### ISSUE TITLE
Load plugins relative to the shareable config they are defined in

### ISSUE THREAD

**Issue by wyze** [eslint/eslint#5501]
**What version of ESLint are you using?**

2.3.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

Espree

**What did you do? Please include the actual source code causing the issue.**

```js
// eslint-config-wyze/index.js
module.exports = {
  extends: [
    'plugin:wyze/recommended',
    'wyze/rules/style'
  ],
  plugins: [
    'wyze'
  ]
}
```

**What did you expect to happen?**

The `eslint-plugin-wyze` to be loaded.

**What actually happened? Please include the actual, raw output from ESLint.**

```shell
  eslint:plugins Failed to load plugin eslint-plugin-wyze. Proceeding without it. +2ms

Oops! Something went wrong! :(

ESLint couldn't find the plugin "eslint-plugin-wyze". This can happen for a couple different reasons:

1. If ESLint is installed globally, then make sure eslint-plugin-wyze is also installed globally. A globally-installed ESLint cannot find a locally-installed plugin.

2. If ESLint is installed locally, then it's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm i eslint-plugin-wyze@latest --save-dev

If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.```
```

----

This is very related to the work to fix #5411 and was wondering if similar logic could be applied to loading plugins as well. This is only an issue on **`npm@2`**. If this is a change not likely to be made, that is fine, I can just have a note in the readme for npm@2 users that they need to install the plugin as a dependency to their project.

I have been changing code around trying to get this working for a couple of hours with some success, but would be willing to help out with a PR if accepted.

**Comment by nzakas** [comment]
No, it can't. Trying to do nested plugins like this leads us into collisions. We'll get around to looking at this at some point, but it's fairly low priority right now. See https://github.com/eslint/eslint/issues/3458 for details.

**Comment by wyze** [comment]
Sorry for the duplicate issue. I should have searched first. I am subscribed to that issue now. Thank you for the response!
